woocommerce/woocommerce
# Add refresh & delete notification tools

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





We previously had issues with Remote Inbox Notifications. When we fixed the problem on our end, it took up to 24 hours for customers to see the fix due to caching. Customers had to install a third-party plugin to refresh the notifications. I propose we add a simple tool to help customers refresh the notifications without needing a third-party plugin. This tool should help the Happiness team guide customers to fix the problem as well.

This PR adds two new tools under WooCommerce -> Status -> Tools to help with Remote Inbox Notification management.

![Screen Shot 2024-06-29 at 12 19 04 PM](https://github.com/woocommerce/woocommerce/assets/4723145/37e80f6e-63d1-4fba-ac62-48137b9e9777)




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. [Create](https://jurassic.ninja/create/?woocommerce-beta-tester&woocommerce) a new JN site with this branch.
2. Go to `Tools -> WCA Test Helper -> Remote Inbox Notification` and click on `Delete all`
3. Go to `WooCommerce -> Status -> Tools`
4. Run `Refresh Remote Inbox Notifications` tool 
5. Go back to `Tools -> WCA Test Helper -> Remote Inbox Notification` and confirm the notifications have been imported back. 
6. Run `Delete Remote Notification` and confirm it deleted the selected notification.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Add Refresh Remote Inbox Notifications and Delete Remote Notification Tools
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
